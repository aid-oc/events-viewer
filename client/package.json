{
  "name": "feast-events-client",
  "version": "0.1.0",
  "private": true,
  "proxy": "http://localhost:5000",
  "dependencies": {
    "@apollo/client": "^3.1.3",
    "@fortawesome/fontawesome-svg-core": "^1.2.30",
    "@fortawesome/free-solid-svg-icons": "^5.14.0",
    "@fortawesome/react-fontawesome": "^0.1.11",
    "graphql": "^15.3.0",
    "prop-types": "^15.7.2",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-scripts": "^3.4.1",
    "styled-components": "^5.1.1"
  },
  "devDependencies": {
    "@testing-library/jest-dom": "^4.2.4",
    "@testing-library/react": "^9.3.2",
    "@testing-library/user-event": "^7.1.2",
    "eslint-config-airbnb": "^18.2.0",
    "eslint-plugin-import": "^2.22.0",
    "eslint-plugin-jsx-a11y": "^6.3.1",
    "eslint-plugin-react": "^7.20.6",
    "eslint-plugin-react-hooks": "^4.1.0"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
