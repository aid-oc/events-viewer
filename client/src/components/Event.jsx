import React from 'react';
import { eventShape, posterShape } from './eventsShape';

const Event = ({
  event, poster,
}) => {
  const {
    description, date, guestCount, type, postcode, address, cancelled, budget, dietary,
  } = event;
  const {
    email, image, name, phone,
  } = poster;
  return (
    <li>
      <h4>
        {`${type} event on ${date} with ${guestCount} guests`}
      </h4>
      <p>{`Budget: ${budget}`}</p>
      <p>{description}</p>
      <p>{`Dietary restrictions: ${dietary}`}</p>
      <p>{`Location: ${postcode}, ${address}`}</p>
      <p>{`Cancelled?: ${cancelled}`}</p>
    </li>
  );
};

Event.propTypes = {
  event: eventShape,
  poster: posterShape,

};

Event.defaultProps = {
  dietary: ['none'],
};

export default Event;
