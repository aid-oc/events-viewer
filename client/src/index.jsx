import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import { ApolloProvider } from '@apollo/client';
import App from './App';
import createClient from './graph/createClient';

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={createClient()}>
      <App />
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root'),
);

if (module.hot) {
  module.hot.accept();
}
