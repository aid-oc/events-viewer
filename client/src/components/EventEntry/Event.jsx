import React from 'react';
import { useMutation } from '@apollo/client';
import { eventShape } from '../eventsShape';
import { EventDetails, EventWarning } from './Event.styles';
import deleteMutation from '../../graph/mutations/deleteEvent';
import getAllEvents from '../../graph/queries/getAllEvents';
import EventInfo from './EventInfo';

const Event = ({
  event,
}) => {
  const {
    id, description, date, guestCount, type,
    postcode, address, cancelled, budget, dietary,
  } = event;
  const [deleteEvent] = useMutation(deleteMutation, {
    variables: { id },
    refetchQueries: [{ query: getAllEvents }],
  });
  return (
    <EventDetails>
      <EventInfo title="Date">
        {date}
      </EventInfo>
      <EventInfo title="Budget">
        {`Â£${budget}`}
      </EventInfo>
      <EventInfo title="Guests">
        {guestCount}
      </EventInfo>
      <EventInfo title="Event Type">
        {type}
      </EventInfo>
      <EventInfo title="Description">
        {description}
      </EventInfo>
      {dietary && dietary.length
      && (
      <EventInfo title="Dietary">
        {dietary.join(', ')}
      </EventInfo>
      )}
      <EventInfo title="Location">
        {`${postcode}, ${address}`}
      </EventInfo>
      {cancelled && (
      <>
        <EventWarning>This event has been cancelled.</EventWarning>
        <button type="button" onClick={() => deleteEvent()}>
          Delete (Actually)
        </button>
      </>
      )}
    </EventDetails>
  );
};

Event.propTypes = {
  event: eventShape.isRequired,
};

export default Event;
