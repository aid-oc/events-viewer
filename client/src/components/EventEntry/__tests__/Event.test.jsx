import React from 'react';
import {
  render,
  fireEvent,
} from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import Event from '../Event';
import deleteEventMutation from '../../../graph/mutations/deleteEvent';

const testEventData = (cancelled = false, dietary = null) => ({
  id: 'someId',
  description: 'someDescription',
  date: '27/08/2020',
  guestCount: 4,
  type: 'someType',
  postcode: 'somePostcode',
  address: 'someAddress',
  budget: 400,
  cancelled,
  dietary,
});

describe('<Event />', () => {
  it('Should match snapshot, shows event data, no dietary or cancellation warning if not provided', () => {
    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <Event event={testEventData()} />
      </MockedProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot, showing cancellation warning if provided', () => {
    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <Event event={testEventData(true)} />
      </MockedProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should match snapshot, showing dietary if provided', () => {
    const { asFragment } = render(
      <MockedProvider mocks={[]}>
        <Event event={testEventData(false, ['vegetarian'])} />
      </MockedProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should perform a mutation without issues preventing a successful render', async () => {
    const mutationMock = [
      {
        request: {
          query: deleteEventMutation,
          variables: { id: 'someId' },
        },
        result: { data: jest.fn(() => ({ id: 'someId' })) },
      },
    ];
    const { getByText, asFragment } = render(
      <MockedProvider mocks={mutationMock}>
        <Event event={testEventData(true)} />
      </MockedProvider>,
    );
    fireEvent.click(getByText('Delete (Actually)'));
    expect(asFragment()).toMatchSnapshot();
  });
});
