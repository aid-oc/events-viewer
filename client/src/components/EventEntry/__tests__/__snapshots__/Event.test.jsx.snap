// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> Should match snapshot, showing cancellation warning if provided 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jfXUMh"
  >
    <p>
      <b>
        Date: 
      </b>
      27/08/2020
    </p>
    <p>
      <b>
        Budget: 
      </b>
      £400
    </p>
    <p>
      <b>
        Guests: 
      </b>
      4
    </p>
    <p>
      <b>
        Event Type: 
      </b>
      someType
    </p>
    <p>
      <b>
        Description: 
      </b>
      someDescription
    </p>
    <p>
      <b>
        Location: 
      </b>
      somePostcode, someAddress
    </p>
    <p
      class="sc-AxirZ jEjlEh"
    >
      This event has been cancelled.
    </p>
    <button
      type="button"
    >
      Delete (Actually)
    </button>
  </div>
</DocumentFragment>
`;

exports[`<Event /> Should match snapshot, showing dietary if provided 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jfXUMh"
  >
    <p>
      <b>
        Date: 
      </b>
      27/08/2020
    </p>
    <p>
      <b>
        Budget: 
      </b>
      £400
    </p>
    <p>
      <b>
        Guests: 
      </b>
      4
    </p>
    <p>
      <b>
        Event Type: 
      </b>
      someType
    </p>
    <p>
      <b>
        Description: 
      </b>
      someDescription
    </p>
    <p>
      <b>
        Dietary: 
      </b>
      vegetarian
    </p>
    <p>
      <b>
        Location: 
      </b>
      somePostcode, someAddress
    </p>
  </div>
</DocumentFragment>
`;

exports[`<Event /> Should match snapshot, shows event data, no dietary or cancellation warning if not provided 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jfXUMh"
  >
    <p>
      <b>
        Date: 
      </b>
      27/08/2020
    </p>
    <p>
      <b>
        Budget: 
      </b>
      £400
    </p>
    <p>
      <b>
        Guests: 
      </b>
      4
    </p>
    <p>
      <b>
        Event Type: 
      </b>
      someType
    </p>
    <p>
      <b>
        Description: 
      </b>
      someDescription
    </p>
    <p>
      <b>
        Location: 
      </b>
      somePostcode, someAddress
    </p>
  </div>
</DocumentFragment>
`;

exports[`<Event /> Should perform a mutation without issues preventing a successful render 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm jfXUMh"
  >
    <p>
      <b>
        Date: 
      </b>
      27/08/2020
    </p>
    <p>
      <b>
        Budget: 
      </b>
      £400
    </p>
    <p>
      <b>
        Guests: 
      </b>
      4
    </p>
    <p>
      <b>
        Event Type: 
      </b>
      someType
    </p>
    <p>
      <b>
        Description: 
      </b>
      someDescription
    </p>
    <p>
      <b>
        Location: 
      </b>
      somePostcode, someAddress
    </p>
    <p
      class="sc-AxirZ jEjlEh"
    >
      This event has been cancelled.
    </p>
    <button
      type="button"
    >
      Delete (Actually)
    </button>
  </div>
</DocumentFragment>
`;
